airbyte-integrations/connectors/source-linnworks/source_linnworks/streams.py

```diff
@@ -70,50 +69,37 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
             }
 
 
-class Location(LinnworksStream):
-    # https://apps.linnworks.net/Api/Method/Locations-GetLocation
-    # Response: StockLocation https://apps.linnworks.net/Api/Class/linnworks-spa-commondata-Locations-ClassBase-StockLocation
+class StockLocations(LinnworksStream):
+    # https://apps.linnworks.net/Api/Method/Inventory-GetStockLocations
+    # Response: List<StockLocation> https://apps.linnworks.net/Api/Class/linnworks-spa-commondata-Inventory-ClassBase-StockLocation
     # Allows 150 calls per minute
     primary_key = "StockLocationIntId"
+    use_cache = True
 
     def path(
         self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
     ) -> str:
-        return "/api/Locations/GetLocation"
-
-    def request_params(
-        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
-    ) -> MutableMapping[str, Any]:
-        return {"pkStockLocationId ": stream_state["pkStockLocationId"]}
+        return "/api/Inventory/GetStockLocations"
 
 
-class StockLocations(LinnworksStream):
-    # https://apps.linnworks.net/Api/Method/Inventory-GetStockLocations
-    # Response: List<StockLocation> https://apps.linnworks.net/Api/Class/linnworks-spa-commondata-Inventory-ClassBase-StockLocation
+class StockLocationDetails(HttpSubStream, StockLocations):
+    # https://apps.linnworks.net/Api/Method/Locations-GetLocation
+    # Response: StockLocation https://apps.linnworks.net/Api/Class/linnworks-spa-commondata-Locations-ClassBase-StockLocation
     # Allows 150 calls per minute
     primary_key = "StockLocationIntId"
 
+    def __init__(self, **kwargs):
+        super().__init__(StockLocations(**kwargs), **kwargs)
+
     def path(
         self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
     ) -> str:
-        return "/api/Inventory/GetStockLocations"
+        return "/api/Locations/GetLocation"
 
-    def read_records(
-        self,
-        sync_mode: SyncMode,
-        cursor_field: List[str] = None,
-        stream_slice: Mapping[str, Any] = None,
-        stream_state: Mapping[str, Any] = None,
-    ) -> Iterable[Mapping[str, Any]]:
-        records = super().read_records(sync_mode, cursor_field, stream_slice, stream_state)
-
-        for record in records:
-            location = Location(authenticator=self.authenticator)
-            stock_location_records = location.read_records(
-                sync_mode, cursor_field, stream_slice, {"pkStockLocationId": record["StockLocationId"]}
-            )
-            record["location"] = next(stock_location_records)
-            yield record
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
```

Comment:
```suggestion
        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any], **kwargs
```
As `stream_slice` is required I don't think you should default it to `None`